BUG: test_umath::test_loss_of_precision[complex256] fails on  ppc64le and aarch64

It seems that `test_umath::test_loss_of_precision[complex256]` is failing on ppc64le and aarch64 on conda-forge.

Not too sure why.

But we had to skip the tests a while back.

Could be due to bug with libc.

We will make CI runs again once 1.17.5 and 1.18.0 are through.

https://github.com/conda-forge/numpy-feedstock/pull/180

Now that travis is back online in conda-forge (allowing us to run on native hardware and not in emulation, which had its own problems), this error still exists on PPC (though not on aarch anymore), cf. https://github.com/conda-forge/numpy-feedstock/pull/268 ([CI run](https://app.travis-ci.com/github/conda-forge/numpy-feedstock/jobs/566559818))

At first glance it looks like a run-of-the-mill tolerance violation, but it's actually pretty substantial (well, in the context of 128bit accuracy at least) - relative error of `4e-19` instead of `<1.2e-30`, so it misses by a cool 11 orders of magnitude. 😅 

```
x = array([1.00000000e-20, 1.21736864e-20, 1.48198641e-20, 1.80412378e-20,
       2.19628372e-20, 2.67368693e-20, 3.254862...5156e-04, 4.54267386e-04,
       5.53010871e-04, 6.73218092e-04, 8.19554595e-04, 9.97700064e-04],
      dtype=float128)
rtol = 1.23259516440783094595582588325435e-30
    def check(x, rtol):
        x = x.astype(real_dtype)
    
        z = x.astype(dtype)
        d = np.absolute(np.arcsinh(x)/np.arcsinh(z).real - 1)
>       assert_(np.all(d < rtol), (np.argmax(d), x[np.argmax(d)], d.max(),
                                  'arcsinh'))
E       AssertionError: (131, 1.54987431981169249551435343964035e-09, 4.0035173453529620614007210953362e-19, 'arcsinh')
```

--- a/numpy/core/tests/test_umath.py
+++ b/numpy/core/tests/test_umath.py
@@ -27,12 +27,17 @@ def on_powerpc():
 
 
 def bad_arcsinh():
-    """The blocklisted trig functions are not accurate on aarch64 for
+    """The blocklisted trig functions are not accurate on aarch64/PPC for
     complex256. Rather than dig through the actual problem skip the
     test. This should be fixed when we can move past glibc2.17
     which is the version in manylinux2014
     """
-    x = 1.78e-10
+    if platform.machine() == 'aarch64':
+        x = 1.78e-10
+    elif on_powerpc():
+        x = 2.16e-10
+    else:
+        return False
     v1 = np.arcsinh(np.float128(x))
     v2 = np.arcsinh(np.complex256(x)).real
     # The eps for float128 is 1-e33, so this is way bigger
@@ -3591,10 +3596,10 @@ class TestComplexFunctions:
         x_basic = np.logspace(-2.999, 0, 10, endpoint=False)
 
         if dtype is np.longcomplex:
-            if (platform.machine() == 'aarch64' and bad_arcsinh()):
+            if bad_arcsinh():
                 pytest.skip("Trig functions of np.longcomplex values known "
-                            "to be inaccurate on aarch64 for some compilation "
-                            "configurations.")
+                            "to be inaccurate on aarch64 and PPC for some "
+                            "compilation configurations.")
             # It's not guaranteed that the system-provided arc functions
             # are accurate down to a few epsilons. (Eg. on Linux 64-bit)
             # So, give more leeway for long complex tests here:

